bolt/lib/Passes/MarkRAStates.cpp

```diff
@@ -0,0 +1,122 @@
+//===- bolt/Passes/MarkRAStates.cpp ---------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements the MarkRAStates class.
+// Three CFIs have an influence on the RA State of an instruction:
+// - NegateRAState flips the RA State,
+// - RememberState pushes the RA State to a stack,
+// - RestoreState pops the RA State from the stack.
+// These were saved as MCAnnotations on instructions they refer to at CFI
```

Comment:
Thanks for adding a detailed description on the passes implementations. I find it very helpful!

Could you kindly reword a bit the first sentence here?
Also in the [PR comment](https://github.com/llvm/llvm-project/pull/120064#issue-2742064203) you may have a bit more info that you could amend here as well?